cabal-install/Distribution/Client/Setup.hs

```diff
@@ -523,7 +524,16 @@ filterConfigureFlags flags cabalLibVersion
       configConstraints = []
       }
 
-    flags_2_1_0 = flags_latest {
+    flags_2_5_0 = flags_latest {
+      -- Cabal < 2.5.0 does not understand --dependency=pkg:COMPONENT=cid
+      --                                   (public sublibraries)
+      configDependencies =
+        let isMainLib CLibName = True
+            isMainLib _        = False
+        in filter (\(_, c, _) -> isMainLib c) $ configDependencies flags
+      }
```

Comment:
I've always thought this filtering was somewhat questionable :) This code shouldn't ever get exercised, but what you have here seems reasonable enough (though, unlikely to lead to a successful build, if Cabal ever thinks it actually needs to pass it through :) (no action needed here)